### PR TITLE
[Feature] Apply AD to parametric observables in PSR routine

### PR DESCRIPTION
**Pull request by dominikandreasseitz**
Fixex https://github.com/pasqal-io/qadence/issues/174

### CODE REVIEW

File: qadence/engines/torch/differentiable_expectation.py [hunk 1]
```diff
@@ -218,18 +226,23 @@ def construct_rules(
...
+
         for obs in observable:
-            for param_id, _ in uuid_to_eigen(obs).items():
-                # We need the embedded fixed params of the observable in the param_values dict
-                # to be able to call expectation. Since torch backward requires
-                # a list of param_ids and values of equal length, we need to pass them to PSR too.
-                # Since they are constants their gradients are 0.
-                param_to_psr[param_id] = lambda x: torch.tensor([0.0], requires_grad=False)
+            for param_id, _ in uuid_to_block(obs.abstract).items():
+                # Trainable parameters in the observable can only be differentiated using AD.
```
Comment (Roland-djee) [inline qadence/engines/torch/differentiable_expectation.py:242]: This is exactly what the method above does right ?

**Review by Roland-djee — APPROVED** [review]
Thanks for this @dominikandreasseitz. Sorry for the late review but for a reason it slipped through my attention.

**Comment by dominikandreasseitz** [comment]
@rajaiitp @Roland-djee , short explanation: PSR works on Gate-Level parameters, not on variational parameters which is the thing youre describing.

**Comment by dominikandreasseitz** [comment]
closing after @jpmoutinho and I agreed on the necessity of a thorough GPSR review before we change the internals of the PSR backward pass

**Pull request closed without merging**